addons/product_images/wizard/product_fetch_image_wizard.py

```diff
@@ -0,0 +1,211 @@
+    nb_product_unprocessable = fields.Integer(string="Number of product unprocessable",
+                                              readonly=True)
+
+    def action_fetch_image(self):
+        """ Trigger a cron job to fetch image on Google
+
+        The 10 first images are immediately fetched to improve the user experience. This way, they
+        can immediately view the images and be assured that the task is running well. Also, if any
+        error occurs, they can be thrown to the user. Then, a cron job is triggered to be run as
+        soon as possible, unless the daily request limit has been reached. In that case, the cron
+        job will be scheduled to run a day later.
+
+        :raises UserError: If the project is misconfigured on Google's side
+        :raises UserError: If the API Key or Search Engine ID is incorrect
+        :return: A notification to inform the user that the task is triggered
+        :rtype: `ir.actions.client`
+        """
+        self.product_processable.image_fetch_pending = True
+        try:
+            self._fetch_image_from_google(self.product_processable[:10])
+        except HTTPError as http_error:
+            if http_error.response.status_code == requests.codes.too_many_requests:
+                self.env.ref('product_images.ir_cron_fetch_image')._trigger(
+                    fields.Datetime.now() + timedelta(days=1.0)
+                )
+                return {
+                    'type': 'ir.actions.client',
+                    'tag': 'display_notification',
+                    'params': {
+                        'type': 'warning',
+                        'title': _("Product images"),
+                        'message': _("Custom search quota exceeded. We will start again tomorrow"),
+                        'next': {'type': 'ir.actions.act_window_close'},
+                    }
+                }
+            elif http_error.response.status_code == requests.codes.forbidden:
+                project_url = re.search(r"(?P<url>https?://[^\s]+)",
+                            http_error.response.json().get('error').get('message')).group('url')
+                raise UserError(_("Custom Search API is not enabled in your Google project. "
+                                  "Enable it by visiting %(url)s then retry. If you enabled this "
+                                  "API recently, wait a few minutes for the action to propagate "
+                                  "to Google's systems and retry.", url=project_url))
+            elif http_error.response.status_code == requests.codes.bad_request:
+                raise UserError(_("Your API Key or your Search Engine ID is incorrect."))
+        except ConnectionError:
+            self.env.ref('product_images.ir_cron_fetch_image')._trigger(
+                fields.Datetime.now() + timedelta(hours=1.0)
+            )
+        else:
+            if self.product_processable[10:]:
```

Comment:
`else`? If there are less than 10 images, say 2, the action will execute silently. This can however become an issue if no matching image is found for either of the 2 products. I believe that it could be good to always display a notification saying "X images were found for Y products". What do you think?

Reply:
I think it could be misleading for the user because we can't display this type of message when it's executed by the cron. He could expect the same message either for 10 or 100 images.

Reply:
This should make my point clearer:
```py
        else:
            message = _(
                "%(matching_image_count)s matching images have been found for %(product_count)s "
                "products.",
                matching_image_count=len(self.products_to_process.filtered('image_1920')),
                product_count=len(self.products_to_process)
            )
            if self.products_to_process[10:]:
                message += '\n' + _("The remaining products will be processed in the background.")
                self.env.ref('product_images.ir_cron_fetch_image')._trigger()
            return {
                'type': 'ir.actions.client',
                'tag': 'display_notification',
                'params': {
                    'type': 'success',
                    'title': _("Product images"),
                    'message': message,
                    'next': {'type': 'ir.actions.act_window_close'},
                }
            }
```